olp-cpp-sdk-core/src/http/winhttp/NetworkWinHttp.cpp

```diff
@@ -233,12 +234,14 @@ NetworkWinHttp::~NetworkWinHttp() {
   std::vector<std::shared_ptr<ResultData>> pending_results;
   {
     std::unique_lock<std::recursive_mutex> lock(mutex_);
-    std::vector<std::shared_ptr<RequestData>> requests_to_cancel;
-    for (const auto& request : http_requests_) {
-      requests_to_cancel.push_back(request.second);
+    std::vector<RequestData*> requests_to_cancel;
```

Comment:
Can we omit creation of additional container and do closing routine inside for loop?